protocol/packages/dex/src/swap.rs

```diff
@@ -6,12 +6,26 @@ use oracle::api::swap::SwapPath;
 use platform::{ica::HostAccount, trx::Transaction};
 use sdk::{cosmos_sdk_proto::Any, cosmwasm_std::StdError};
 
+#[cfg(feature = "testing")]
+pub struct SwapRequest<GIn>
+where
+    GIn: Group,
+{
+    pub token_in: CoinDTO<GIn>,
+    pub swap_path: SwapPath,
+}
+
 pub trait ExactAmountIn {
+    #[cfg(feature = "testing")]
+    fn parse_request<GIn>(request: Any) -> SwapRequest<GIn>
+    where
+        GIn: Group;
+
```

Comment:
the rule of ordering here was 'production/test' - first, 'build/parse' - second
so this fn should be move accordingly